### ISSUE TITLE
As an admin I want TEI transcription content migrated to IIIF annotation so that I can manage and edit it in the new transcription editor.

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/geniza#914]
## testing notes
All TEI content has been migrated to the new annotation format and should be editable in the new transcription editor.  

Not sure how to test this thoroughly (and some of it will be tested under other issues), but let's start with these:
- [x] number of documents with transcriptions in public search should be similar to production
- [ ] number of documents with digital editions in admin search should be similar to production
- [x]  footnotes linking the transcription to the document should be updated as follows:
   - [x] converted from edition to digital edition
   - [x] provisional content removed
   - [x] if existing edition footnote also had a link or other document relations, it should have been preserved and a new digital edition footnote created
- [x] content linked to correct scholarship record (uses same basic logic as sync transcription, so has some similar limitations)
   - [x] check at least one document with more than one transcription
   - [x] check at least one document where transcription for a join is split across two TEI files
- [x] transcription chunked into editable sections by label and grouped with pages based on label (uses same logic as sync transcription script)
   - see https://github.com/Princeton-CDH/geniza/issues/913 for checking editable version
 - [ ] spot check transcriptions that were reported as having unicode normalization to see if there are any concerns (expectation: changes should not be visible)

## dev notes
Adapt from existing sync tei script to the degree possible (maybe early prototype iiif annotation code also helpful). Should create labeled blocks similar to what we have now in the tei -> html; attach translations to the full canvas. (Use % notation if annotorious allows. Possibly tag as full-image/unblocked or similar.) Needs to handle documents without images, with mixed images, etc.

next steps:
- [x] map source uri to dc:source on the annotation
- [x] revise tei to html to use standard `<li>` for easy editing in tinymce; try this: if an unnumbered list item occurs, map it to a paragraph instead, and if the number is not 1 set the start attribute (will need to test if this displays properly and is easily editable in tinymce)
- [x] migrate footnotes to digital editions

after first-round testing:
- [x] normalize unicode when migrating; report on affected files
- [x] empty content has ""; investigate and resolve (migration or display issue)
- [x] migration script should report on how many footnotes with content still remain
- [x] investigate why so many footnotes with content still remain

**Comment by rlskoeser** [comment]
output from running the script in staging:

```
Processed 4492 TEI file(s).
Created 9411 annotation(s).
Could not find footnotes for 63 documents:
	/srv/www/geniza/data/tei_xml/4638.xml
	/srv/www/geniza/data/tei_xml/5440.xml
	/srv/www/geniza/data/tei_xml/5422.xml
	/srv/www/geniza/data/tei_xml/5485.xml
	/srv/www/geniza/data/tei_xml/5459.xml
	/srv/www/geniza/data/tei_xml/5456.xml
	/srv/www/geniza/data/tei_xml/5465.xml
	/srv/www/geniza/data/tei_xml/5446.xml
	/srv/www/geniza/data/tei_xml/4618.xml
	/srv/www/geniza/data/tei_xml/4530.xml
	/srv/www/geniza/data/tei_xml/4715.xml
	/srv/www/geniza/data/tei_xml/5462.xml
	/srv/www/geniza/data/tei_xml/5519.xml
	/srv/www/geniza/data/tei_xml/5387.xml
	/srv/www/geniza/data/tei_xml/5353.xml
	/srv/www/geniza/data/tei_xml/5409.xml
	/srv/www/geniza/data/tei_xml/5436.xml
	/srv/www/geniza/data/tei_xml/5378.xml
	/srv/www/geniza/data/tei_xml/5413.xml
	/srv/www/geniza/data/tei_xml/4737.xml
	/srv/www/geniza/data/tei_xml/2926.xml
	/srv/www/geniza/data/tei_xml/5445.xml
	/srv/www/geniza/data/tei_xml/5342.xml
	/srv/www/geniza/data/tei_xml/1437.xml
	/srv/www/geniza/data/tei_xml/5404.xml
	/srv/www/geniza/data/tei_xml/5437.xml
	/srv/www/geniza/data/tei_xml/5500.xml
	/srv/www/geniza/data/tei_xml/5482.xml
	/srv/www/geniza/data/tei_xml/4623.xml
	/srv/www/geniza/data/tei_xml/5421.xml
	/srv/www/geniza/data/tei_xml/5389.xml
	/srv/www/geniza/data/tei_xml/5438.xml
	/srv/www/geniza/data/tei_xml/5525.xml
	/srv/www/geniza/data/tei_xml/5393.xml
	/srv/www/geniza/data/tei_xml/5451.xml
	/srv/www/geniza/data/tei_xml/5344.xml
	/srv/www/geniza/data/tei_xml/5352.xml
	/srv/www/geniza/data/tei_xml/5452.xml
	/srv/www/geniza/data/tei_xml/4671.xml
	/srv/www/geniza/data/tei_xml/4727.xml
	/srv/www/geniza/data/tei_xml/4646.xml
	/srv/www/geniza/data/tei_xml/4722.xml
	/srv/www/geniza/data/tei_xml/5416.xml
	/srv/www/geniza/data/tei_xml/5406.xml
	/srv/www/geniza/data/tei_xml/4609.xml
	/srv/www/geniza/data/tei_xml/5457.xml
	/srv/www/geniza/data/tei_xml/5441.xml
	/srv/www/geniza/data/tei_xml/5526.xml
	/srv/www/geniza/data/tei_xml/5357.xml
	/srv/www/geniza/data/tei_xml/5454.xml
	/srv/www/geniza/data/tei_xml/5492.xml
	/srv/www/geniza/data/tei_xml/4680.xml
	/srv/www/geniza/data/tei_xml/5418.xml
	/srv/www/geniza/data/tei_xml/5458.xml
	/srv/www/geniza/data/tei_xml/5460.xml
	/srv/www/geniza/data/tei_xml/4535.xml
	/srv/www/geniza/data/tei_xml/4613.xml
	/srv/www/geniza/data/tei_xml/5412.xml
	/srv/www/geniza/data/tei_xml/5403.xml
	/srv/www/geniza/data/tei_xml/5334.xml
	/srv/www/geniza/data/tei_xml/4728.xml
	/srv/www/geniza/data/tei_xml/4552.xml
	/srv/www/geniza/data/tei_xml/4622.xml
```

**Comment by kseniaryzhova** [comment]
@rlskoeser On the admin test site, the amount of documents with digital editions in 4295 while on the production admin it's 4359 which is the same as the test front end.

**Comment by kseniaryzhova** [comment]
@rlskoeser In the contents of the footnote, the transcription content is gone but there was this symbol left behind: "" (PGPID 6176 admin site). Not sure if it's significant.

**Comment by rlskoeser** [comment]
@kseniaryzhova thanks for noting, we must be either not cleaning it up properly or not displaying empty content right! Will investigate.

I just did a quick query on the test site and I count 380 footnotes with content still attached — that's a lot more than I was expecting. I'll make a note to add a report/check to the script and also to look into these. I'm guessing there must be a mix of content to be cleaned up and content to be migrating, but maybe it also means we're not migrating some cases correctly (i.e. documents with multiple editions).

**Comment by kseniaryzhova** [comment]
@rlskoeser while looking at joins, we found the transcription editor was displaying oddly on the public test site. I'm on a PC, using Microsoft Edge with adblocker. @richmanrachel is having the same issue on her Mac on Chrome.
![image](https://user-images.githubusercontent.com/105385128/187277720-0e52d9a7-c47b-4b4b-809b-94c7e7f1a6ce.png)

**Comment by kseniaryzhova** [comment]
Note: @rlskoeser 9117 was the only document we tested where the transcription was split across two TEI files. May merit more testing? We're having trouble finding the right thing to test.

**Comment by rlskoeser** [comment]
> @rlskoeser while looking at joins, we found the transcription editor was displaying oddly on the public test site. I'm on a PC, using Microsoft Edge with adblocker. @richmanrachel is having the same issue on her Mac on Chrome. ![image](https://user-images.githubusercontent.com/105385128/187277720-0e52d9a7-c47b-4b4b-809b-94c7e7f1a6ce.png)

Ah, I see! this was when you switched to the second transcription, right? I see the problem, thanks for documenting it.
https://test-geniza.cdh.princeton.edu/en/documents/9116/

**Comment by richmanrachel** [comment]
@rlskoeser - also getting these weird bars when trying to edit one of the transcriptions:
<img width="959" alt="image" src="https://user-images.githubusercontent.com/68028846/187278715-2c6578fc-7a2f-49ba-8bf8-c188e349ac88.png">

**Comment by rlskoeser** [comment]
> Note: @rlskoeser 9117 was the only document we tested where the transcription was split across two TEI files. May merit more testing? We're having trouble finding the right thing to test.

@richmanrachel are there any examples from that spreadsheet of transcriptions we were cleaning up that would be helpful? I think there were a few cases like this, it would be good to check what's happening with those

**Comment by rlskoeser** [comment]
> @rlskoeser - also getting these weird bars when trying to edit one of the transcriptions ...

@richmanrachel what browser are you using, and what operating system? I think that one is rendering ok for me on Firefox...

**Comment by richmanrachel** [comment]
@rlskoeser - I'm on Chrome.

Trying to use examples from our transcription cleanup but running into some problems. I'll message on Slack b/c it's beyond this issue

**Comment by rlskoeser** [comment]
@richmanrachel ok, thanks — maybe we will need to make a separate issue to track migration and testing for transcription handling of these joins; it's ok with me if you want to move on and test other parts of the transcription editing that aren't as complicated!

**Comment by richmanrachel** [comment]
@rlskoeser - I think we're almost ready to close this issue. Just trying to summarize next steps:
1. Make new issue for tracking transcription joins/migration/editing
2. Check in later to see whether re-syncing made the numbers the same for front and backend of test site
3. Follow up on content that stayed in the Content box of footnotes.

**Comment by rlskoeser** [comment]
> @rlskoeser In the contents of the footnote, the transcription content is gone but there was this symbol left behind: "" (PGPID 6176 admin site). Not sure if it's significant.

I did some testing on this, no matter what empty value I set the cleared out content to (empty string, null value, empty dict) it displays in the content box. I think it's fine to treat it as a display issue, since what we care about is that the content has been removed (and the empty content field that's left will be removed in the update after the transcription migration).

**Comment by rlskoeser** [comment]
🤔 hmm, maybe? I wonder if the solr index is out of sync.

I'm working on some revisions to the migration script (normalizing unicode, reporting on unmigrated footnotes with content). Once that is done, I can synchronize production to test so they match exactly and then re-run the tei to annotation migration.

**Comment by rlskoeser** [comment]
I added some logic to the migration script to report on unmigrated footnotes (edition footnotes with transcription content), and will share the output when we run in staging. I did some spot checking on the ones being reported in my dev environment, and what I'm seeing seems reasonable.

Here are some notes on the ones I checked:


- https://geniza.princeton.edu/admin/corpus/document/646/change/
 two Goitein edition footnotes that are slightly different in metadata; is edition content the same?
- https://geniza.princeton.edu/admin/corpus/document/5384/change/
  had an edition footnote with old "see other" content; I cleaned up in production
- https://geniza.princeton.eduadmin/corpus/document/2179/change/
   https://geniza.princeton.edu/admin/footnotes/footnote/4962/change/
  There is an Oded footnote with content, but I don't see an Oded transcription in the TEI; my guess is that this is transcription content that was synced to the wrong footnote
- PGPID 5466 — this is one of the cases where we know there are two TEI files that need to be combined; 
  It looks like we have two sets of content attached to the two images, which is what I expect — hopefully the transcription editor will make it easier to rearrange this; may need to clean up a redundant footnote after migration

**Comment by richmanrachel** [comment]
@rlskoeser - for 646 the content is the same. Should I just copy the "with corrections" note and put it into the one that seems to be syncing and delete the other footnote?

For 2179, I'm seeing the note that Oded's is "awaiting digitization" meaning that it might be in a google doc rather than bitbucket

Yes, you're right about 5466 (but there were also duplicate IB footnotes for ח21 - I just deleted one)

**Comment by rlskoeser** [comment]
@richmanrachel any of the cleanup we can do now is good! Everything you've said here sounds right to me.

Let me get you the full list after I run the script on the test site.

**Comment by rlskoeser** [comment]
Here is the summary output from the script:

```
Processed 4492 TEI file(s).
Created 9547 annotation(s).
Could not find footnotes for 1 document:
	/srv/www/geniza/data/tei_xml/5440.xml
Normalized unicode for 122 documents:
	/srv/www/geniza/data/tei_xml/4590.xml
	/srv/www/geniza/data/tei_xml/9118.xml
	/srv/www/geniza/data/tei_xml/9070.xml
	/srv/www/geniza/data/tei_xml/4573.xml
	/srv/www/geniza/data/tei_xml/9080.xml
	/srv/www/geniza/data/tei_xml/9096.xml
	/srv/www/geniza/data/tei_xml/4579.xml
	/srv/www/geniza/data/tei_xml/9141.xml
	/srv/www/geniza/data/tei_xml/9121.xml
	/srv/www/geniza/data/tei_xml/9131.xml
	/srv/www/geniza/data/tei_xml/4600.xml
	/srv/www/geniza/data/tei_xml/9126.xml
	/srv/www/geniza/data/tei_xml/9092.xml
	/srv/www/geniza/data/tei_xml/9060.xml
	/srv/www/geniza/data/tei_xml/4612.xml
	/srv/www/geniza/data/tei_xml/9122.xml
	/srv/www/geniza/data/tei_xml/9103.xml
	/srv/www/geniza/data/tei_xml/9065.xml
	/srv/www/geniza/data/tei_xml/9107.xml
	/srv/www/geniza/data/tei_xml/4642.xml
	/srv/www/geniza/data/tei_xml/9136.xml
	/srv/www/geniza/data/tei_xml/9078.xml
	/srv/www/geniza/data/tei_xml/3567.xml
	/srv/www/geniza/data/tei_xml/4641.xml
	/srv/www/geniza/data/tei_xml/4608.xml
	/srv/www/geniza/data/tei_xml/9112.xml
	/srv/www/geniza/data/tei_xml/9054.xml
	/srv/www/geniza/data/tei_xml/4622.xml
	/srv/www/geniza/data/tei_xml/4515.xml
	/srv/www/geniza/data/tei_xml/9145.xml
	/srv/www/geniza/data/tei_xml/4592.xml
	/srv/www/geniza/data/tei_xml/4613.xml
	/srv/www/geniza/data/tei_xml/5283.xml
	/srv/www/geniza/data/tei_xml/4645.xml
	/srv/www/geniza/data/tei_xml/9120.xml
	/srv/www/geniza/data/tei_xml/4623.xml
	/srv/www/geniza/data/tei_xml/4671.xml
	/srv/www/geniza/data/tei_xml/9109.xml
	/srv/www/geniza/data/tei_xml/4584.xml
	/srv/www/geniza/data/tei_xml/9059.xml
	/srv/www/geniza/data/tei_xml/5269.xml
	/srv/www/geniza/data/tei_xml/4587.xml
	/srv/www/geniza/data/tei_xml/9073.xml
	/srv/www/geniza/data/tei_xml/4621.xml
	/srv/www/geniza/data/tei_xml/4680.xml
	/srv/www/geniza/data/tei_xml/9144.xml
	/srv/www/geniza/data/tei_xml/4644.xml
	/srv/www/geniza/data/tei_xml/4632.xml
	/srv/www/geniza/data/tei_xml/9067.xml
	/srv/www/geniza/data/tei_xml/4628.xml
	/srv/www/geniza/data/tei_xml/9111.xml
	/srv/www/geniza/data/tei_xml/9100.xml
	/srv/www/geniza/data/tei_xml/4633.xml
	/srv/www/geniza/data/tei_xml/4524.xml
	/srv/www/geniza/data/tei_xml/9066.xml
	/srv/www/geniza/data/tei_xml/4595.xml
	/srv/www/geniza/data/tei_xml/9091.xml
	/srv/www/geniza/data/tei_xml/4744.xml
	/srv/www/geniza/data/tei_xml/9094.xml
	/srv/www/geniza/data/tei_xml/4601.xml
	/srv/www/geniza/data/tei_xml/4730.xml
	/srv/www/geniza/data/tei_xml/9081.xml
	/srv/www/geniza/data/tei_xml/4519.xml
	/srv/www/geniza/data/tei_xml/4580.xml
	/srv/www/geniza/data/tei_xml/9057.xml
	/srv/www/geniza/data/tei_xml/3489.xml
	/srv/www/geniza/data/tei_xml/1025.xml
	/srv/www/geniza/data/tei_xml/4610.xml
	/srv/www/geniza/data/tei_xml/9071.xml
	/srv/www/geniza/data/tei_xml/9076.xml
	/srv/www/geniza/data/tei_xml/4659.xml
	/srv/www/geniza/data/tei_xml/9143.xml
	/srv/www/geniza/data/tei_xml/4643.xml
	/srv/www/geniza/data/tei_xml/4609.xml
	/srv/www/geniza/data/tei_xml/4620.xml
	/srv/www/geniza/data/tei_xml/4629.xml
	/srv/www/geniza/data/tei_xml/4606.xml
	/srv/www/geniza/data/tei_xml/9058.xml
	/srv/www/geniza/data/tei_xml/9056.xml
	/srv/www/geniza/data/tei_xml/4655.xml
	/srv/www/geniza/data/tei_xml/3562.xml
	/srv/www/geniza/data/tei_xml/4572.xml
	/srv/www/geniza/data/tei_xml/4652.xml
	/srv/www/geniza/data/tei_xml/9101.xml
	/srv/www/geniza/data/tei_xml/4663.xml
	/srv/www/geniza/data/tei_xml/4679.xml
	/srv/www/geniza/data/tei_xml/1440.xml
	/srv/www/geniza/data/tei_xml/4650.xml
	/srv/www/geniza/data/tei_xml/9115.xml
	/srv/www/geniza/data/tei_xml/4578.xml
	/srv/www/geniza/data/tei_xml/4662.xml
	/srv/www/geniza/data/tei_xml/4625.xml
	/srv/www/geniza/data/tei_xml/4536.xml
	/srv/www/geniza/data/tei_xml/4647.xml
	/srv/www/geniza/data/tei_xml/4676.xml
	/srv/www/geniza/data/tei_xml/9047.xml
	/srv/www/geniza/data/tei_xml/9130.xml
	/srv/www/geniza/data/tei_xml/4537.xml
	/srv/www/geniza/data/tei_xml/9063.xml
	/srv/www/geniza/data/tei_xml/5299.xml
	/srv/www/geniza/data/tei_xml/9140.xml
	/srv/www/geniza/data/tei_xml/4583.xml
	/srv/www/geniza/data/tei_xml/5291.xml
	/srv/www/geniza/data/tei_xml/4576.xml
	/srv/www/geniza/data/tei_xml/4615.xml
	/srv/www/geniza/data/tei_xml/9110.xml
	/srv/www/geniza/data/tei_xml/9062.xml
	/srv/www/geniza/data/tei_xml/9068.xml
	/srv/www/geniza/data/tei_xml/4666.xml
	/srv/www/geniza/data/tei_xml/4673.xml
	/srv/www/geniza/data/tei_xml/4646.xml
	/srv/www/geniza/data/tei_xml/9044.xml
	/srv/www/geniza/data/tei_xml/9085.xml
	/srv/www/geniza/data/tei_xml/4656.xml
	/srv/www/geniza/data/tei_xml/9114.xml
	/srv/www/geniza/data/tei_xml/4667.xml
	/srv/www/geniza/data/tei_xml/9043.xml
	/srv/www/geniza/data/tei_xml/9104.xml
	/srv/www/geniza/data/tei_xml/4752.xml
	/srv/www/geniza/data/tei_xml/9125.xml
	/srv/www/geniza/data/tei_xml/9046.xml
	/srv/www/geniza/data/tei_xml/9123.xml

21 unmigrated footnotes
	Edition of ENA 4020.65 (PGPID 646)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/1330/change/
	Edition and Translation of T-S 10J32.6 + T-S 10J9.24 (PGPID 4718)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2583/change/
	Edition and Translation of T-S 8J6.8 + T-S 13J30.3 (PGPID 2179)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/4962/change/
	Edition of T-S Misc.25.132 + T-S Misc.25.139 (PGPID 1372)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5657/change/
	Edition of ENA NS 21.14 (PGPID 2691)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/1467/change/
	Edition of T-S 10J15.16 (PGPID 2861)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2233/change/
	Edition of T-S Ar.30.46 + T-S Ar.30.89 (PGPID 3884)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5103/change/
	Edition of CUL Add.3414.1–2 (PGPID 5384)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/15170/change/
	Edition of T-S NS 184.50 + T-S NS 184.58 +  T-S NS 184.62 + T-S NS 184.70-.72 + T-S NS 184.98 (PGPID 5371)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14220/change/
	Edition of NLI 577.3/6 (PGPID 5499)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14271/change/
	Edition of CUL Or.1080 J95 (PGPID 5555)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14304/change/
	Edition of Bodl. MS heb. d 66/64–65 (PGPID 5340)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14196/change/
	Edition of Bodl. MS heb. d 66/64–65 (PGPID 5340)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14195/change/
	Edition of T-S Ar.41.81 (PGPID 5540)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14295/change/
	Edition of T-S 18J4.18 + T-S NS J21 (PGPID 4732)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14147/change/
	Edition of T-S 8J5.8 (PGPID 34347)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/13860/change/
	Edition of T-S NS 321.79 + T-S 13J4.5 (PGPID 4273)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5915/change/
	Edition of T-S 13J27.15 + T-S 8J13.2 (PGPID 3392)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/3464/change/
	Edition of T-S NS 321.79 + T-S 13J4.5 (PGPID 4273)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/3588/change/
	Edition of T-S 10J24.7 + T-S 10J29.4 (PGPID 2971)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2400/change/
	Edition of T-S 13J27.15 + T-S 8J13.2 (PGPID 3392)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/4350/change/
```


duplicating the footnote so the links are clickable:

*	Edition of ENA 4020.65 (PGPID 646)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/1330/change/
*	Edition and Translation of T-S 10J32.6 + T-S 10J9.24 (PGPID 4718)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2583/change/
*	Edition and Translation of T-S 8J6.8 + T-S 13J30.3 (PGPID 2179)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/4962/change/
*	Edition of T-S Misc.25.132 + T-S Misc.25.139 (PGPID 1372)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5657/change/
*	Edition of ENA NS 21.14 (PGPID 2691)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/1467/change/
*	Edition of T-S 10J15.16 (PGPID 2861)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2233/change/
*	Edition of T-S Ar.30.46 + T-S Ar.30.89 (PGPID 3884)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5103/change/
*	Edition of CUL Add.3414.1–2 (PGPID 5384)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/15170/change/
*	Edition of T-S NS 184.50 + T-S NS 184.58 +  T-S NS 184.62 + T-S NS 184.70-.72 + T-S NS 184.98 (PGPID 5371)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14220/change/
*	Edition of NLI 577.3/6 (PGPID 5499)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14271/change/
*	Edition of CUL Or.1080 J95 (PGPID 5555)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14304/change/
*	Edition of Bodl. MS heb. d 66/64–65 (PGPID 5340)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14196/change/
*	Edition of Bodl. MS heb. d 66/64–65 (PGPID 5340)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14195/change/
*	Edition of T-S Ar.41.81 (PGPID 5540)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14295/change/
*	Edition of T-S 18J4.18 + T-S NS J21 (PGPID 4732)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/14147/change/
*	Edition of T-S 8J5.8 (PGPID 34347)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/13860/change/
*	Edition of T-S NS 321.79 + T-S 13J4.5 (PGPID 4273)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/5915/change/
*	Edition of T-S 13J27.15 + T-S 8J13.2 (PGPID 3392)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/3464/change/
*	Edition of T-S NS 321.79 + T-S 13J4.5 (PGPID 4273)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/3588/change/
*	Edition of T-S 10J24.7 + T-S 10J29.4 (PGPID 2971)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/2400/change/
*	Edition of T-S 13J27.15 + T-S 8J13.2 (PGPID 3392)	https://test-geniza.cdh.princeton.edu/admin/footnotes/footnote/4350/change/

**Comment by richmanrachel** [comment]
@rlskoeser - for 2179, I am actually seeing the content of Oded's transcription when I open the footnote. But you couldn't find the TEI file for it?

**Comment by richmanrachel** [comment]
Just got this error message when trying to click into one of the links for India Book for 5499:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/68028846/187996290-5542819f-deab-4a87-8ac4-a24a8d199291.png">
I think I fixed the other issue with this document, but wanted to mention this

**Comment by rlskoeser** [comment]
> @rlskoeser - for 2179, I am actually seeing the content of Oded's transcription when I open the footnote. But you couldn't find the TEI file for it?

TEI 2179.xml ms description says it is from Goitein's typed texts. The only old PGPID I see for 2179 is 1119, and there is a 1119.xml but the source description says it's Gil.

**Comment by richmanrachel** [comment]
@rlskoeser - should I just uncheck "edition" for Oded for now on 2179?

I was able to solve most of the other problems. The outlying ones fall into two main camps:
1. I don't see a problem with: 2691, 2861, 5540, 13860, 4732
    The commonality between some of them is that there's a Goitein TEI + a url to his papers, but I didn't know how to fix 
    things beyond merging the footnotes that were the same/deleting duplicates.
2. Needs more attention (with merging transcriptions and/or more serious combing through): 5340, 2179, 3392, 5555

**Comment by rlskoeser** [comment]
@richmanrachel thanks for this careful review.  Yes, let's uncheck edition on that Oded footnote.

Based on what you've checked, does it seem like the migration script is doing the right thing and these are just footnotes that we need to clean up or deal with? It could easily be that some of them aren't real problems, the query I used is not exact (I think at least one of the ones flagged looked like one where I'd cleaned up the content manually after the last sync transcription change). The reason I wanted to report on them is to help us make sure we've migrated all of the existing transcriptions, and this seemed like one way of catching potential problems or omissions.

**Comment by richmanrachel** [comment]
@rlskoeser - yes, if the "no problems" aren't problems then we should be good! I need to figure out how to deal with the harder ones, but I think they should wait till after migration so I can merge the transcriptions. 

Is this ready to close, then?!

**Comment by rlskoeser** [comment]
@richmanrachel 🤯 that would be amazing if the migration is in that good shape.

How do the total numbers compare this time? And would you spot check some of the records where I normalized the unicode? I think after that we can close. 

Do we still need an issue to handle joins and multi-edition docs? I didn't create one yet because I wasn't sure what exactly we need to do!

**Comment by richmanrachel** [comment]
@rlskoeser - Production frontend and backend agree with have 4,360 transcriptions. Test site has 4,358 on both ends. Is that close enough?

I'm still seeing unsupported characters amongst the normalized unicode ones (ex. https://test-geniza.cdh.princeton.edu/en/documents/9118/) but was the normalization just to fix the footnotes?

**Comment by rlskoeser** [comment]
@richmanrachel I think that's probably close enough on the totals! (although I wish I knew where the discrepancy was coming from)

The record you flagged with the display problems did not show up as needing to be normalized.

Normalization puts the unicode into the canonical form for the characters (since there are multiple ways to write many characters in unicode). I don't _expect_ it to break anything, the changes should be invisible, but I thought it would be good for someone to look who knows what the transcriptions ought to look like — so I would consider that test as passing!

**Comment by richmanrachel** [comment]
@rlskoeser - great, closing then!
